strawberryfields/tdm/tdmprogram.py

```diff
@@ -303,6 +304,98 @@ def context(self, *args, copies=1, shift="default"):
         self.loop_vars = self.params(*[f"p{i}" for i in range(len(args))])
         return self
 
+    def compile(self, *, device=None, compiler=None):
+        """Compile the time-domain program given a Strawberry Fields photonic hardware device specification.
+        At this stage the compilation is simply a check that the program matches the device.
```

Comment:
Just curious: for general CV circuits we'd have separate compilers. Was the motivation behind adding the logic here for simplicity (only have one type of compilation for TDM & it's just a check)?

Reply:
The motivation, as far as I can tell, is that for TDM programs the only compilation that one can hope to do is to check that the program matches exactly the device. There (at least as of now) clever tricks to figure out if one can map it or not.